packages/nextjs/src/utils/config/webpack.ts

```diff
@@ -0,0 +1,206 @@
+import { getSentryRelease } from '@sentry/node';
+import { dropUndefinedKeys, logger } from '@sentry/utils';
+import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
+
+import {
+  BuildContext,
+  EntryPropertyObject,
+  ExportedNextConfig,
+  SentryWebpackPluginOptions,
+  WebpackConfigFunction,
+  WebpackConfigObject,
+  WebpackEntryProperty,
+} from './types';
+import {
+  SENTRY_CLIENT_CONFIG_FILE,
+  SENTRY_SERVER_CONFIG_FILE,
+  SERVER_SDK_INIT_PATH,
+  storeServerConfigFileLocation,
+} from './utils';
+
+export { SentryWebpackPlugin };
+
+// TODO: merge default SentryWebpackPlugin ignore with their SentryWebpackPlugin ignore or ignoreFile
+// TODO: merge default SentryWebpackPlugin include with their SentryWebpackPlugin include
+// TODO: drop merged keys from override check? `includeDefaults` option?
+
+const defaultSentryWebpackPluginOptions = dropUndefinedKeys({
+  url: process.env.SENTRY_URL,
+  org: process.env.SENTRY_ORG,
+  project: process.env.SENTRY_PROJECT,
+  authToken: process.env.SENTRY_AUTH_TOKEN,
+  configFile: 'sentry.properties',
+  stripPrefix: ['webpack://_N_E/'],
+  urlPrefix: `~/_next`,
+  include: '.next/',
+  ignore: ['.next/cache', 'server/ssr-module-cache.js', 'static/*/_ssgManifest.js', 'static/*/_buildManifest.js'],
+});
+
+/**
+ * Construct the function which will be used as the nextjs config's `webpack` value.
+ *
+ * Sets:
+ *   - `devtool`, to ensure high-quality sourcemaps are generated
+ *   - `entry`, to include user's sentry config files (where `Sentry.init` is called) in the build
+ *   - `plugins`, to add SentryWebpackPlugin (TODO: optional)
+ *
+ * @param userNextConfig The user's existing nextjs config, as passed to `withSentryConfig`
+ * @param userSentryWebpackPluginOptions The user's SentryWebpackPlugin config, as passed to `withSentryConfig`
+ * @returns The function to set as the nextjs config's `webpack` value
+ */
+export function constructWebpackConfigFunction(
+  userNextConfig: ExportedNextConfig = {},
+  userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+): WebpackConfigFunction {
+  const newWebpackFunction = (config: WebpackConfigObject, options: BuildContext): WebpackConfigObject => {
+    // if we're building server code, store the webpack output path as an env variable, so we know where to look for the
+    // webpack-processed version of `sentry.server.config.js` when we need it
+    if (config.target === 'node') {
+      storeServerConfigFileLocation(config);
```

Comment:
This is the only use of the function, imo this becomes easier to read if you just move the function body into the conditional here.

Reply:
I disagree. That's what it was originally and I pulled it out into its own function. I think of it like zooming in. At this level, I don't actually care _how_ it's doing that, just _that_ it's doing it.

(Besides, wherever the code to do this lives, it's likely not long for this world.)